lib/active_shipping/carriers/usps.rb

```diff
@@ -238,9 +238,9 @@ def extract_event_details(node)
       
       timestamp = "#{node.at('EventDate').text}, #{node.at('EventTime').text}"
       event_code = node.at('EventCode').text
-      city = node.at('EventCity').text
-      state = node.at('EventState').text
-      zip_code = node.at('EventZIPCode').text
+      city = try_text(node.at('EventCity'))
```

Comment:
I wasn't sure that method was defined since this isn't a rails app. TIL active_shipping depends on activesupport. Thanks!